dasharo-security/tpm-support.robot: Refactor TPM version and support …

SebastianCzapla opened the pull request:
…tests

Currently, TPM Support and TPM Version test are split into different tests groups. Merging of version and support checks into one allows us to skip few extra reboot cycles during testing. Additionally, it provides opportunity to add coverage for TPM 1.2

SebastianCzapla committed:
dasharo-security/tpm-support.robot: Refactor TPM version and support tests

This commit introduces two new variables, EXPECTED_TPM_CHIP and
EXPECTED_TPM_VERSION. Additionally, refactor few keywords and tests
within tpm-support.robot

Signed-off-by: Sebastian Czapla <sebastian.czapla@3mdeb.com>

miczyg1 commented:
It is not entirely redundant. See: https://github.com/Dasharo/open-source-firmware-validation/issues/495#issuecomment-2358194086

The test to verify TPM version in firmware should check if it is TPM 1.2 or 2.0

SebastianCzapla commented:
This branch was rebased onto this https://github.com/Dasharo/open-source-firmware-validation/pull/487
Here's a log from dasharo-security test suite run with this patch applied.

[dasharo-security.tar.gz](https://github.com/user-attachments/files/17130422/dasharo-security.tar.gz)

miczyg1 commented on dasharo-security/tpm-support.robot:
```diff
@@ -34,8 +34,8 @@ TPM001.001 TPM Support (firmware)
     Get Cbmem From Cloud
-    ${out}=    Execute Command In Terminal    cbmem -L
-    Should Contain    ${out}    TPM2 log
+    ${out}=    Execute Command In Terminal    cbmem -1 | grep -E 'TPM2 detected| TCPA log'
```
Previously it was `cbmem -L` not cbmem `-1`. Not sure if this will work as intended.

miczyg1 replied:
Also there is a space before `TCPA log` which may affect the grep

SebastianCzapla replied:
I have experienced segfault with ```cbmem -L``` on TPM 1.2 platforms.

Here's an ending of cbmem -L log:
```
TCPA log entry 16:
	PCR: 2127990177
	Event type: Unknown (0xc8a02c0e >= 19)
	Digest: 4b204bba40ce87b9d818753c5f8d08181f480e6f
	Event data: 
7022cce803e7262d8852d03c0d0d21d80b28bd303b8b3a1b8649db978858e3f26abdbe9fba5cfa70667c936c2c2531cc0322876c26a5348c0102ee5592b96e822c054cdfc3fcc62078545d9c827d7ef9e58a8864e8014bc7ff15fa1351ba979a11[...]
```

The log actually continues as 300kB+ dump of random numbers, which ends at some point with a segfault.
Here's a little info that dmesg gives:
```
[ 1499.410096] cbmem[5238]: segfault at 7b35459d2000 ip 0000603bad38cdeb sp 00007fff858301f0 error 4 in cbmem[603bad38b000+4000] likely on CPU 2 (core 2, socket 0)
[ 1499.410111] Code: 3d f6 ff ff 48 8d 0d 14 58 00 00 ba ab 05 00 00 48 8d 35 e7 22 00 00 48 8d 3d 8d 24 00 00 e8 dc e5 ff ff 31 c9 31 c0 41 89 ce <0f> b6 54 05 00 4c 89 ee bf 01 00 00 00 31 c0 e8 a1 e6 ff ff 41 8d
``

cbmem -1 works no problem on both 1.2 and 2.0 platforms.

Within cbmem -1 logs, I have found a point which could be checked against for these TPM support cases:
For TPM 1.2 (coreboot TPM 1.2 measurements)
```
[DEBUG]  TPM: Digest of `CBFS: fallback/payload` to PCR 2 measured
[DEBUG]  Checking segment from ROM address 0xffc608ac
[DEBUG]  Checking segment from ROM address 0xffc608c8
[DEBUG]  Loading segment from ROM address 0xffc608ac
[DEBUG]    code (compression=1)
[DEBUG]    New segment dstaddr 0x00800000 memsize 0xe00000 srcaddr 0xffc608e4 filesize 0x151bb6
[DEBUG]  Loading Segment: addr: 0x00800000 memsz: 0x0000000000e00000 filesz: 0x0000000000151bb6
[DEBUG]  using LZMA
[DEBUG]  Loading segment from ROM address 0xffc608c8
[DEBUG]    Entry Point 0x00800850
[DEBUG]  BS: BS_PAYLOAD_LOAD run times (exec / console): 1312 / 83 ms
[INFO ]  coreboot TPM 1.2 measurements:

[INFO ]   PCR-2 cf11730b6290e3ad42e0753f436edfe1612ec0fc SHA1 [FMAP: FMAP]
[INFO ]   PCR-2 801df4236e0c7bdd1f980c4b45c7c19cc18282d8 SHA1 [CBFS: bootblock]
[INFO ]   PCR-2 c1922cd3de966dc40257073496ae306ee0ffa1f6 SHA1 [CBFS: cmos.default]
[INFO ]   PCR-2 9baee5f388d903d5b72e20697209cfa67a724e44 SHA1 [CBFS: fallback/romstage]
[INFO ]   PCR-2 fd2b3cf1bd64b3928ffc916d7799c76e5fa96de0 SHA1 [CBFS: cmos_layout.bin]
[INFO ]   PCR-2 9cc3514487871b14663e7ba29efc6fec5fed7c76 SHA1 [CBFS: sch5545_ecfw.bin]
[INFO ]   PCR-2 48d7b91079795aae8b7082b76c10e3ca03520bad SHA1 [CBFS: fallback/postcar]
[INFO ]   PCR-2 8efffbbf487c06d4328ce4fbba2314554f2a2925 SHA1 [CBFS: cpu_microcode_blob.bin]
[INFO ]   PCR-2 ea3f3ba2449d59b7cf13ee645d640ba6a018acad SHA1 [CBFS: vbt.bin]
[INFO ]   PCR-2 931a1ae3cd1cb7c815b2ebfd57bb6508e15d5edf SHA1 [CBFS: txt_bios_acm.bin]
[INFO ]   PCR-2 b6e1865a9e3127b0ea67bdff6dbf33dd0235314b SHA1 [CBFS: txt_sinit_acm.bin]
[INFO ]   PCR-2 931a1ae3cd1cb7c815b2ebfd57bb6508e15d5edf SHA1 [CBFS: txt_bios_acm.bin]
[INFO ]   PCR-2 d599814817d3b807dfb4abc462b4606fac62b4c9 SHA1 [CBFS: fallback/dsdt.aml]
[INFO ]   PCR-2 fd2b3cf1bd64b3928ffc916d7799c76e5fa96de0 SHA1 [CBFS: cmos_layout.bin]
[INFO ]   PCR-2 d765e221f11f8c0233e5807ce37ae83fe6705346 SHA1 [CBFS: fallback/payload]

[DEBUG]  BS: BS_PAYLOAD_BOOT entry times (exec / console): 0 / 128 ms
[DEBUG]  ICH-NM10-PCH: watchdog disabled
[DEBUG]  Jumping to boot code at 0x00800850(0x7fb48000)
```

and for TPM 2.0

```
[DEBUG]  using LZMA
[SPEW ]  [ 0x00800000, 01600000, 0x01600000) <- ff86c590
[DEBUG]  Loading segment from ROM address 0xff86c574
[DEBUG]    Entry Point 0x00800860
[SPEW ]  Loaded segments
[DEBUG]  BS: BS_PAYLOAD_LOAD run times (exec / console): 380 / 0 ms
[INFO ]  coreboot skipped calling FSP notify phase: 00000040.
[INFO ]  coreboot skipped calling FSP notify phase: 000000f0.
[INFO ]  coreboot TCPA measurements:

[INFO ]   PCR-2 1be8f33e0df7790491fa72960da3ed0ebbe73362c2803cf014169837e3fe097e SHA256 [FMAP: FMAP]
[INFO ]   PCR-2 51cf039e2ca217e8718625815929e3d053287160a08d8f3ef380631a356546cf SHA256 [CBFS: bootblock]
[INFO ]   PCR-2 32a677bcb45f59b452250d2692ba0fd5623fae9d49899868f3260b2e27bcbb0a SHA256 [CBFS: fallback/romstage]
[INFO ]   PCR-2 5e8e90f12f4804625a0937d91fbe65210fdb60a0989e318cf91288a4b2bc1086 SHA256 [CBFS: fspm.bin]
[INFO ]   PCR-2 9cc5f1dadf78d07a3879a756b9d4a8f3f0e7b269bc6b8a12ad4da329c1e79c4d SHA256 [CBFS: fallback/postcar]
[INFO ]   PCR-2 a3f0752ce30dbadd95e5c16e85f29a339c049bea506db905ae1a086b94b68895 SHA256 [CBFS: fallback/ramstage]
[INFO ]   PCR-2 90fe3367f6d5b14a7e4362a9bda859f302946db9dacbb02e240195969caecb1f SHA256 [CBFS: cpu_microcode_blob.bin]
[INFO ]   PCR-2 5bf36bb2f18b2623c16dd48977e761d185d4bfe34aa87e636250990982ee1233 SHA256 [CBFS: fsps.bin]
[INFO ]   PCR-2 59e0fad23a73f0c5765206f3be5521825ff26850587c18268b7617ff4078f515 SHA256 [CBFS: vbt.bin]
[INFO ]   PCR-2 f034cbb27726e2f59a75115a822b1ec989548811361280ad0c3a064ebd30fea4 SHA256 [CBFS: fallback/dsdt.aml]
[INFO ]   PCR-2 97650b169e466d377f62d6f6ca7b6ffd6a881b46645b6e7a2ff65ef890a746e5 SHA256 [CBFS: fallback/payload]

[DEBUG]  Finalizing chipset.
[DEBUG]  apm_control: Finalizing SMM.
```

Do these coreboot tpm xx measurement lines make sense as targets for grep?

miczyg1 replied:
> I have experienced segfault with cbmem -L on TPM 1.2 platforms.

Looks like a bug in cbmem or coreboot

miczyg1 replied:
> Do these coreboot TPM/TCPA measurement lines make sense as targets for grep?

Relying on either `coreboot TPM 1.2 measurements` or `coreboot TCPA measurements` is not the right choice.

First, it is merely a print which may say anything, but it doesn't say which format is used or what TPM is really used. It just says what it tried to extend the PCRs with some values.

Secondly TCPA measurements and TPM1.2 measurements is equivalent per TPM specification.

If anything one could use the other prints to check for TPM version:

1. UEFI Payload prints  `TpmVersion: 0x2` or `Tcg2ConfigPeimEntryPoint: TPM2 detected` when a TPM2.0 is connected.
2. coreboot prints `Found TPM SLB9670 TT 2.0 by Infineon` on laptops. But on any other platform it may be different depending on the TPM chip used. So it is not a generic method of detecting TPM version.
3. When using UEFI Payload, one may enter setup then enter Device Manager and then check which TP msetup is present:
- for TPM 2.0 it will be `TCG2 Configuration`
- for TPM 1.2 it will be `TCG Configuration`

SebastianCzapla replied:
Changed the detection to use Device Manager to look for menu entries.

miczyg1 commented on dasharo-security/tpm-support.robot:
```diff
@@ -121,7 +114,7 @@ TPM003.002 Check TPM Physical Presence Interface (Ubuntu)
     Switch To Root User
     ${out}=    Execute Command In Terminal    cat /sys/class/tpm/tpm0/ppi/version
-    Should Contain    ${out}    1.3
+    Should Contain Any    ${out}    1.2    1.3
```
I don't think it should be any.

If the expected TPM version is 2, then PPI version is 1.3. For TPM1.2 the PPI will be 1.2

miczyg1 commented on dasharo-security/tpm-support.robot:
```diff
@@ -138,3 +131,40 @@ TPM003.003 Check TPM Physical Presence Interface (Windows)
+        Should Contain    ${out}    sha256:
+    ELSE IF    '${tpm_ver}' == '1'
+        Detect Or Install Package    tpm-tools
+        ${out}=    Execute Command In Terminal    tpm_selftest
```
If we validate TPm by checking PCRs, we should also do it the same way for TPM1.2. TPm1.2 PCRs are also populated in sysfs `/sys/class/tpm/tpm0/device/pcrs`

krystian-hebel replied:
TPM 2.0 also reports PCRs in sysfs for quite some time, in different format, but the same way works for TPM 1.2 (except it has only `sha1` obviously):

```bash
$ ls /sys/class/tpm/tpm0/pcr-sha*
/sys/class/tpm/tpm0/pcr-sha1:
0  10  12  14  16  18  2   21  23  4  6  8
1  11  13  15  17  19  20  22  3   5  7  9

/sys/class/tpm/tpm0/pcr-sha256:
0  10  12  14  16  18  2   21  23  4  6  8
1  11  13  15  17  19  20  22  3   5  7  9
$ cat /sys/class/tpm/tpm0/pcr-sha256/4
E61988A7E3DD2C2F69C3226E4904E69D8A43BA0A35E3BC0930A8A9E0E7CAAD21
```

miczyg1 replied:
Yup, valid point. No need to install tpm2-tools just for this

SebastianCzapla commented:
Updated and rebased as per suggestions.
Using Setup Menu to detect TPM has a tradeoff, it no longer is automated test for laptop testing. Is this an acceptable tradeoff? @miczyg1 

Also, when testing this solution, I found an issue with OptiPlex platform https://github.com/Dasharo/dasharo-issues/issues/1091

filipleple commented:
Overall seems good, managed to get it to pass on OptiPlex with SeaBIOS (minus the network-related fails, which are a [separate issue](https://github.com/Dasharo/open-source-firmware-validation/issues/628))

```
==============================================================================
Tpm-Support
==============================================================================
TPM001.001 TPM Support (firmware) :: This test aims to verify that... ......
Chip detection failed, attempting cbmem log detection

TPM001.001 TPM Support (firmware) :: This test aims to verify that... | FAIL |
Keyword 'Check Internet Connection On Linux' failed after retrying 5 times. The last error was: 'ping: g
oogle-public-dns-a.google.com: Temporary failure in name resolution' does not contain ', 0% packet loss'
------------------------------------------------------------------------------
TPM001.002 TPM Support (Ubuntu) :: Check whether the TPM is initia... | FAIL |
AttributeError: module 'requests.exceptions' has no attribute 'ProtocolError'
------------------------------------------------------------------------------
TPM001.003 TPM Support (Windows) :: Check whether the TPM is initi... | SKIP |
TPM001.003 not supported
------------------------------------------------------------------------------
TPM002.001 Verify TPM version (firmware) :: This test aims to veri... ......
Chip detection failed, attempting cbmem log detection

TPM002.001 Verify TPM version (firmware) :: This test aims to veri... | FAIL |
Keyword 'Check Internet Connection On Linux' failed after retrying 5 times. The last error was: 'ping: g
oogle-public-dns-a.google.com: Temporary failure in name resolution' does not contain ', 0% packet loss'
------------------------------------------------------------------------------
TPM002.002 Verify TPM version (Ubuntu) :: This test aims to verify... | PASS |
------------------------------------------------------------------------------
TPM002.003 Verify TPM version (Windows) :: This test aims to verif... | SKIP |
TPM002.003 not supported
------------------------------------------------------------------------------
TPM003.001 Check TPM Physical Presence Interface (firmware) :: Thi... | FAIL |
Keyword 'Check Internet Connection On Linux' failed after retrying 5 times. The last error was: 'ping: g
oogle-public-dns-a.google.com: Temporary failure in name resolution' does not contain ', 0% packet loss'
------------------------------------------------------------------------------
TPM003.002 Check TPM Physical Presence Interface (Ubuntu) :: This ... | PASS |
------------------------------------------------------------------------------
TPM003.003 Check TPM Physical Presence Interface (Windows) :: This... | SKIP |
TPM003.003 not supported
------------------------------------------------------------------------------
Since the asset has been checkout manually by you prior running this script, it will NOT be checked in a
utomatically. Please return the device when work is finished.
Tpm-Support                                                           | FAIL |
9 tests, 2 passed, 4 failed, 3 skipped
==============================================================================
```

pietrushnic commented:
@macpijan, does this PR get some benefits of https://github.com/Dasharo/open-source-firmware-validation/pull/574 ?

SebastianCzapla commented on dasharo-security/tpm-support.robot:
```diff
@@ -30,22 +30,24 @@ TPM001.001 TPM Support (firmware)
     ...    correctly and the PCRs can be accessed from the firmware.
```
I guess because we are basing the test on cbmem, which down the line relies on firmware, it qualifies as a firmware based detection, but I agree, that documentation does not reflect the mechanism well. Do you think documentation change is enough?

filipleple replied:
I think the `(Firmware)` and `(Ubuntu)` labels are most useful to quickly see if a test is firmware UI dependent or not, e.g., when determining a test scope. As in - it goes through EDK2 menus, won't work on SeaBIOS/heads and laptops which don't offer a serial connection. But I guess changing the documentation to clearly state it goes straight into Ubuntu should do the trick

SebastianCzapla commented:
[dasharo-security.zip](https://github.com/user-attachments/files/18094503/dasharo-security.zip)

Rebased and tested on Protectli vp3230

SebastianCzapla commented on dasharo-security/tpm-support.robot:
```diff
@@ -30,22 +30,24 @@ TPM001.001 TPM Support (firmware)
     Skip If    not ${TESTS_IN_UBUNTU_SUPPORT}    TPM001.001 not supported
     Power On
-    Boot System Or From Connected Disk    ubuntu
+    Menus.Boot System Or From Connected Disk    ubuntu
```
Different PR fixed issues I had, so I deleted the Menus. prefix.

miczyg1 reviewed (changes requested):
The tests will stop working on laptops right? AS per https://github.com/Dasharo/open-source-firmware-validation/pull/507#issuecomment-2416057360

Would be best to check cbmem log for a particular TPM model printed by coreboot (not edk2)

miczyg1 commented on dasharo-security/tpm-support.robot:
```diff
@@ -26,16 +26,18 @@ Suite Teardown      Run Keyword
-    Get Cbmem From Cloud
-    ${out}=    Execute Command In Terminal    cbmem -L
-    Should Contain Any    ${out}    TPM2 log    TCPA log
+    ${result}=    Run Keyword And Ignore Error    Validate Expected TPM Via Cbmem Chip
```
```suggestion
    ${result}=    Run Keyword And Ignore Error    Validate Expected TPM Chip Via Cbmem
```

Is the gramatically correct name

miczyg1 replied:
Also how is that different from `Validate Expected TPM Via Cbmem Log`? The names are so similar but they seem to do different things...

SebastianCzapla replied:
@miczyg1 While at name changes, should I also move the keywords to lib/tpm.robot?

miczyg1 replied:
Not sure what should go to libs and what not. Maybe @macpijan can say more

miczyg1 commented on dasharo-security/tpm-support.robot:
```diff
@@ -138,18 +147,33 @@ TPM003.003 Check TPM Physical Presence Interface (Windows)
+    ${tpm_chip_found}=    Execute Command In Terminal    cbmem -1 | grep -i "Found TPM"
+    Should Contain    ${tpm_chip_found}    ${TPM_EXPECTED_CHIP}
+
+Validate Expected TPM Via Cbmem Log
```
```suggestion
Validate Expected TPM Via TPM Log in Cbmem
```

miczyg1 commented on dasharo-security/tpm-support.robot:
```diff
@@ -138,18 +147,33 @@ TPM003.003 Check TPM Physical Presence Interface (Windows)
+        Fail    Platform TPM version mismatch
+    END
+
+Validate Expected TPM Via Cbmem Chip
```
```suggestion
Validate Expected TPM Chip Via Cbmem
```

miczyg1 commented on dasharo-security/tpm-support.robot:
```diff
@@ -69,9 +71,11 @@ TPM002.001 Verify TPM version (firmware)
-    Get Cbmem From Cloud
-    ${out}=    Execute Command In Terminal    cbmem -L
-    Should Contain Any    ${out}    TPM2 log    TCPA log
+    ${result}=    Run Keyword And Ignore Error    Validate Expected TPM Via Cbmem Chip
```
```suggestion
    ${result}=    Run Keyword And Ignore Error    Validate Expected TPM Chip Via Cbmem
```

SebastianCzapla committed:
scripts/get-robot-variables.sh: Add TPM to config generation

Signed-off-by: Sebastian Czapla <sebastian.czapla@3mdeb.com>

SebastianCzapla committed:
dasharo-security/tpm2-commands.robot: Add more skip conditions

Signed-off-by: Sebastian Czapla <sebastian.czapla@3mdeb.com>

SebastianCzapla committed:
scripts/get-robot-variables.sh: Fix awk expression

Signed-off-by: Sebastian Czapla <sebastian.czapla@3mdeb.com>

krystian-hebel commented on dasharo-security/measured-boot.robot:
```diff
@@ -301,8 +301,10 @@ Measured Boot Suite Setup
     Prepare Test Suite
     Skip If    not ${MEASURED_BOOT_SUPPORT}    Measured boot is not supported
     Skip If    not ${TESTS_IN_UBUNTU_SUPPORT}    Tests in Ubuntu are not supported
+    Skip If    '${TPM_EXPECTED_VERSION}' == '0'    TPM Version in platform config does not support this suite
```
Why not `Skip If    not ${TPM_SUPPORT}`? Alternatively, both config options could be merged, with `TPM_EXPECTED_VERSION` equal to 0 (or even `none` because of dynamic typing) meaning what `TPM_SUPPORT = False` currently is used for.

SebastianCzapla replied:
Merged the variables into "TPM_SUPPORTED_VERSION"

krystian-hebel commented on dasharo-security/tpm2-commands.robot:
```diff
@@ -264,9 +265,11 @@ TPM2 Suite Setup
     Power On
     Boot System Or From Connected Disk    ubuntu
     Login To Linux
+    Verify Presence Of TPM Via Sysfs
```
In other cases this is done after switching to root.

krystian-hebel commented on dasharo-security/tpm-support.robot:
```diff
@@ -20,22 +21,27 @@ Suite Setup         Run Keywords
     Power On
     Boot System Or From Connected Disk    ubuntu
     Login To Linux
     Switch To Root User
-    Get Cbmem From Cloud
-    ${out}=    Execute Command In Terminal    cbmem -L
-    Should Contain Any    ${out}    TPM2 log    TCPA log
+    Verify Presence Of TPM Via Sysfs
```
These keywords in this order are repeated in almost all of the tests, perhaps it would make sense to extract them to a local keyword.

SebastianCzapla replied:
Code move into "Prepare TPM Test On Ubuntu" kwd.

SebastianCzapla committed:
tpm-support: Merge TPM variables

Signed-off-by: Sebastian Czapla <sebastian.czapla@3mdeb.com>

SebastianCzapla committed:
dasharo-security/tpm-support.robot: Extract common actions into kwd

Signed-off-by: Sebastian Czapla <sebastian.czapla@3mdeb.com>

krystian-hebel commented on platform-configs/protectli-vp6650.robot:
```diff
@@ -28,3 +28,5 @@ ${PLATFORM_RAM_SIZE}=           65536
 ...                             64-62-66-22-84-f7
 ...                             64-62-66-22-84-f8
+
+${TPM_SUPPORTED_VERSION}=       ${None}
```
Same as default, not necessary.

krystian-hebel commented on dasharo-security/tpm-support.robot:
```diff
@@ -149,6 +124,16 @@ TPM003.003 Check TPM Physical Presence Interface (Windows)
 #    [Documentation]    This test aims to verify that the TPM Physical Presence
 #    ...    Interface is working properly in the firmware by changing active TPM PCR banks.
-#    Skip If    not ${tpm_support}    TPM003.004 not supported
+#    Skip If    not ${TPM_SUPPORTED_VERSION}    TPM003.004 not supported
```
Even though this is commented, it still should be `Skip If    ${TPM_SUPPORTED_VERSION} == None`.

SebastianCzapla committed:
dasharo-security/tpm-support.robot: Fix skip conditions in WIP code

Signed-off-by: Sebastian Czapla <sebastian.czapla@3mdeb.com>

miczyg1 commented on lib/tpm.robot:
```diff
@@ -67,3 +67,40 @@ Validate PCRs Against Event Log
+    ${tpm_chip_found}=    Execute Command In Terminal    cbmem -1 | grep -i "Found TPM"
+    Should Contain    ${tpm_chip_found}    ${TPM_EXPECTED_CHIP}
+
+Validate Expected TPM Version Via Cbmem Log
```
```suggestion
Validate Expected TPM Version Via Cbmem TPM Eventlog
```

miczyg1 commented on lib/tpm.robot:
```diff
@@ -67,3 +67,40 @@ Validate PCRs Against Event Log
+        Fail    Platform TPM version mismatch
+    END
+
+Validate Expected TPM Chip Via Cbmem
```
```suggestion
Validate Expected TPM Chip Via Cbmem Console Log
```

miczyg1 reviewed (changes requested):
Still don't quite agree on the names of the keywords. They both check logs in cbmem, but doesn't explicitly say which log.

SebastianCzapla committed:
dasharo-security/tpm-support.robot: Clarify kwd names, fix variable gen

Signed-off-by: Sebastian Czapla <sebastian.czapla@3mdeb.com>

SebastianCzapla commented:
@miczyg1 Fixed the names as per suggestion. Additionally, I've introduced small change to get-robot-variables.sh; I changed the usage of path to usage of a variable that holds this path, as it was also missing. Retested after all the changes, here's a [log](https://github.com/user-attachments/files/18657889/dasharo-security.zip)

miczyg1 commented on scripts/get-robot-variables.sh:
```diff
@@ -196,7 +196,7 @@ else
 
-if [ -f /usr/local/bin/cbmem ]; then
+if [ -f ${CBMEM_BINARY_PATH} ]; then
     TPM_EXPECTED_CHIP=$(sudo /usr/local/bin/cbmem -1 | grep "Found TPM" | awk 'NR==1{print $6}');
```
The command does not use `CBMEM_BINARY_PATH`

SebastianCzapla replied:
Fixed

miczyg1 commented:
@SebastianCzapla it just needs a rebase before it can be merged fast-forward

SebastianCzapla commented:
@miczyg1 Done
